src/dune/lib.ml

```diff
@@ -1542,7 +1568,17 @@ module Compile = struct
       t.requires
     in
     let lib_deps_info =
-      let pps = Lib_info.pps t.info in
+      let pps =
+        (* CHECK *)
+        let db = Option.value_exn db in
```

Comment:
This is the line I am not sure about.

Reply:
Just had a look, the only call site for `for_lib` looks like this:

```ocaml
      let t = Option.some_if (not allow_overlaps) t in
      Compile.for_lib t lib
```

so I'd say we instead pass `allow_overlaps` to `for_lib` and avoid the `Option.value_exn` altogether

Reply:
Of course! Amended as suggested. I don't have anything else left on my list.

Reply:
Indeed, looks good to merge!